zerver/lib/cache.py

```diff
@@ -714,6 +714,18 @@ def flush_stream(sender: Any, **kwargs: Any) -> None:
         cache_delete(bot_dicts_in_realm_cache_key(stream.realm))
 
 
+def flush_stream_post_delete(sender: Any, **kwargs: Any) -> None:
+    from zerver.models import UserProfile
+
+    stream = kwargs["instance"]
+    cache_delete(get_stream_cache_key(stream.name, stream.realm_id))
+
+    if UserProfile.objects.filter(
+        Q(default_sending_stream=stream) | Q(default_events_register_stream=stream)
+    ).exists():
+        cache_delete(bot_dicts_in_realm_cache_key(stream.realm))
```

Comment:
I would have expected the logic to be more or less identical before; at least the 

`kwargs.get("update_fields") is None` case above should have triggered this logic.

Is the issue that we should delete from the cache rather than set a new value in it?  I think that makes sense, but could be done as `if kwargs.get("update_fields") is None`  without forking this function?

As a sidenote, do you have an easy reproducer for the bug / how confident are you that this change fixed it?  Those details could go in the commit message.

Reply:
> Is the issue that we should delete from the cache rather than set a new value in it? 

Yup. That was the issue.

> I think that makes sense, but could be done as if kwargs.get("update_fields") is None without forking this function?

Okay. I have removed the extra function.

> As a sidenote, do you have an easy reproducer for the bug / how confident are you that this change fixed it? Those details could go in the commit message.

I have included more details in the commit message.